code/game/objects/items/documents.dm

```diff
@@ -115,3 +221,218 @@
+		print_time = 5 SECONDS
+		print_sound_mode = INSPECTOR_PRINT_SOUND_MODE_CLASSIC
+		power_per_print = INSPECTOR_POWER_USAGE_NORMAL
+
+/obj/item/inspector/clown/bananium/attackby(obj/item/I, mob/user, params)
+	..()
+	check_settings_legality()
+	if(istype(I, /obj/item/paper/fake_report)||(paper_charges>=max_paper_charges))
+		to_chat(user, "<span class='info'>\The [src] refuses to consume \the [I]!</span>")
+	else if(istype(I, /obj/item/paper))
+		to_chat(user, "<span class='info'>\The [src] consumes \the [I]!</span>")
+		paper_charges+=charges_per_paper
+		if(paper_charges>max_paper_charges)
+			paper_charges = max_paper_charges
+		qdel(I)
+
+/obj/item/inspector/clown/bananium/Initialize()
+	. = ..()
+	playsound(src, 'sound/effects/angryboat.ogg', 150, FALSE)
+
+/obj/item/inspector/clown/bananium/create_slip()
+	if(print_time == 0.1 SECONDS)
+		var/obj/item/paper/fake_report/water/slip = new(get_turf(src))
+		slip.generate_report(get_area(src))
+	else
+		..()
+
+/obj/item/inspector/clown/bananium/print_report(mob/user)
+	if(print_time != 0.1 SECONDS)
+		..()
+	else if(paper_charges == 0)
+		if(cell.use(power_to_speak))
+			say("ERROR! OUT OF PAPER! MAXIMUM PRINTING SPEED UNAVAIBLE! SWITCH TO A SLOWER SPEED TO OR PROVIDE PAPER!")
+		else
+			to_chat(user, "<span class='info'>\The [src] doesn't seem to be on... Perhaps it ran out of power?")
+	else
+		paper_charges--
+		..()
+
+/obj/item/inspector/clown/bananium/cycle_print_time(mob/user)
+	var/message = "You turn the screw-like dial, setting the device's scanning speed to "
+	if(print_time == 0.1 SECONDS)
+		power_per_print = INSPECTOR_POWER_USAGE_NORMAL
+		print_time = 5 SECONDS
+		message += "SLOW."
+	else if(print_time == 5 SECONDS)
+		print_time = 1 SECONDS
+		message += "LIGHTNING FAST."
+	else
+		print_time = 0.1 SECONDS
+		power_per_print = INSPECTOR_POWER_USAGE_HONK
+		message += "HONK!"
+	balloon_alert(user, message)
+
+/**
+ * Reports printed by fake N-spect scanner
+ *
+ * Not valid for the bounty.
+ */
+/obj/item/paper/fake_report
+	name = "encrypted station inspection"
+	desc = "Contains no information about the station's current status."
+	icon = 'icons/obj/bureaucracy.dmi'
+	icon_state = "slip"
+	///What area the inspector scanned when the report was made. Used to generate the examine text of the report
+	var/area/scanned_area
+	show_written_words = FALSE
+
+/obj/item/paper/fake_report/proc/generate_report(area/scan_area)
+	scanned_area = scan_area
+	icon_state = "slipfull"
+
+	var/list/characters = list()
+	characters += GLOB.alphabet
+	characters += GLOB.alphabet_upper
+	characters += GLOB.numerals
+
+	var/length = rand(23, 123)
+	var/i
+	for(i = 0; i<length; i++)
+		if(prob(90))
+			info += pick_list_replacements(CLOWN_NONSENSE_FILE, "honk")
+		else if(prob(1))
+			info += pick_list_replacements(CLOWN_NONSENSE_FILE, "rare")
+		else
+			info += pick_list_replacements(CLOWN_NONSENSE_FILE, "non-honk-clown-words")
```

Comment:
```suggestion
	var/new_info = list()
	for(var/i in 1 to length)
		var/roll = rand(0, 1000)
		switch(roll)
			if(0 to 900)
				new_info += pick_list_replacements(CLOWN_NONSENSE_FILE, "honk")
			if(901 to 999)
				new_info += pick_list_replacements(CLOWN_NONSENSE_FILE, "non-honk-clown-words")
			if(1000)
				new_info += pick_list_replacements(CLOWN_NONSENSE_FILE, "rare")
	info += new_info.Join()
```
* The C-like for loop will check the value of the index every loop. Prefer the list loop unless you modify the index.
* The rand way is just another way to do it. Feel free to ignore it if you think it's less clear or something. But it's good to know about (the switch ranges and whatnot).
* The list + Join() method is actually more performant because BYOND string handling is kind of really bad.